pkg/test/framework/components/mixer/attrmanifest.go

```diff
@@ -0,0 +1,52 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package mixer
+
+import (
+	"errors"
+	"strings"
+
+	"istio.io/istio/pkg/test/framework/repository"
+	"istio.io/istio/pkg/test/framework/scopes"
+	"istio.io/istio/pkg/test/helm"
+)
+
+// Extract attribute manifest from Helm charts.
+func extractAttributeManifestFromHelm() (string, error) {
+	// We don't care about deploymentName, namespace or values file, or other settings, as we only
+	// want to get attribute manifest for the local environment.
+	s, err := helm.Template(
+		"attributemanifest",
+		"istio-system",
+		repository.IstioChartDir(),
+		"", nil)
+	if err != nil {
+		return "", err
+	}
+
+	// Split the template into chunks
+	parts := strings.Split(s, "\n---\n")
```

Comment:
Is this notation specific to k8s, or is it a general yaml separator? We have a lot of this code lying around ... would be good to use a utility for this.

Reply:
Yaml specific: http://yaml.org/spec/1.0/ (section 2.2).
Moved to the top-level "test" library.